lrpar/src/lib/ctbuilder.rs

```diff
@@ -692,29 +742,34 @@ where
                 // Call the user code
                 let args = (0..grm.prod(pidx).len())
                     .map(|i| format!("{prefix}arg_{i}", prefix = ACTION_PREFIX, i = i + 1))
+                    .chain(grm.param_args().iter().map(|arg| arg.0.clone().to_string()))
                     .collect::<Vec<_>>();
                 // If the rule `r` that we're calling has the unit type then Clippy will warn that
                 // `enum::A(wrapper_r())` is pointless. We thus have to split it into two:
                 // `wrapper_r(); enum::A(())`.
                 match grm.actiontype(ridx) {
                     Some(s) if s == "()" => {
                         outs.push_str(&format!("\n        {prefix}action_{pidx}({prefix}ridx, {prefix}lexer, {prefix}span, {args});
-        {actionskind}::{actionskindprefix}{ridx}(())",
+        ({actionskind}::{actionskindprefix}{ridx}(()), {param_pat})",
                             actionskind = ACTIONS_KIND,
                             actionskindprefix = ACTIONS_KIND_PREFIX,
                             prefix = ACTION_PREFIX,
                             ridx = usize::from(ridx),
                             pidx = usize::from(pidx),
-                            args = args.join(", ")));
+                            args = args.join(", "),
+                            param_pat = self.param_pattern(grm),
+                        ));
                     }
                     _ => {
-                        outs.push_str(&format!("\n        {actionskind}::{actionskindprefix}{ridx}({prefix}action_{pidx}({prefix}ridx, {prefix}lexer, {prefix}span, {args}))",
+                        outs.push_str(&format!("\n        ({actionskind}::{actionskindprefix}{ridx}({prefix}action_{pidx}({prefix}ridx, {prefix}lexer, {prefix}span, {args})), {param_pat})",
```

Comment:
So, I need to look at this more carefully, as it seems like a very tricky part.

`{args}`, now contains param_args, from the chain added above, but then went out of scope in the action call.
It seems quite possible that this could lead to a dramatic amount of clean up in figuring out when/where we can stop threading this parameter through.  But I haven't done that detailed of a review of things yet.